dashboard/views.py

```diff
@@ -259,10 +259,39 @@ def grade_distribution(request):
     df_grade['grade_avg'] = average_grade
     return HttpResponse(df_grade.to_json(orient='records'))
 
+def assignment_progress(request):
+    logger.info(assignment_view.__name__)
+    sql = "select assignment_id,local_graded_date as graded_date,score,name,assign_grp_name,local_date as due_date,points_possible,group_points,weight from (" \
+          "(select assignment_id,local_graded_date,score from" \
+          "(select id from USER where sis_name = %(current_user)s ) as u join" \
+          "(select user_id,assignment_id,local_graded_date,score from SUBMISSION) as sub on sub.user_id=u.id) as rock join" \
+          "(select assign_id,name,assign_grp_name,local_date,points_possible,group_points,weight from" \
+          "(select id as assign_id,assignment_group_id, local_date,name,points_possible from ASSIGNMENT where course_id = %(course_id)s) as a join" \
+          "(select id,name as assign_grp_name,group_points, weight from ASSIGNMENT_GROUPS) as ag on ag.id=a.assignment_group_id) as bottom on rock.assignment_id = bottom.assign_id)"
+    df = pd.read_sql(sql,conn,params={"current_user": current_user,'course_id': UDW_COURSE_ID},parse_dates={'due_date': '%Y-%m-%d','graded_date':'%m/%d/%Y'})
```

Comment:
We probably should start putting some more error checking in here incase there is no result returned. Maybe at least in a try/catch. If df is empty it's going to cause problems.

Reply:
Or maybe just something like what's already used

    # return if there is no data during this interval
    if (df.empty):
        return HttpResponse("no data")

Reply:
Done